ingestion/src/metadata/ingestion/source/storage/s3/connection.py

```diff
@@ -59,8 +59,15 @@ def test_connection(
     of a metadata workflow or during an Automation Workflow
     """
 
+    def test_buckets(connection: S3Connection, client: S3ObjectStoreClient):
+        if connection.bucketNames:
+            return partial(
+                client.s3_client.list_objects, Bucket=connection.bucketNames[0]
```

Comment:
does it mean we'll only test the connection for the 1st bucket in the list?

Reply:
yes